scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -1729,6 +1732,18 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
           Type.Macro(macroSplice())
         case Ident("?") if dialect.allowQuestionMarkPlaceholder =>
           next(); atPos(in.prevTokenPos, auto)(Type.Placeholder(typeBounds()))
+        case ident: Ident
+            if (ident.value == "+" || ident.value == "-") &&
+              ahead(token.is[Underscore]) &&
+              allowPlusMinusUnderscore =>
+          autoPos {
+            accept[Ident]
+            accept[Underscore]
+            if (dialect.allowPlusMinusUnderscoreAsPlaceholder)
+              Type.Placeholder(typeBounds())
+            else
+              Type.Name(s"${ident.value}_")
```

Comment:
does this version not allow type bounds?

Reply:
yep, but it is in line with scala/scala#9605